### PR TITLE
Take a lock while printing to the console/UART

### PR DESCRIPTION
**Pull request by erikcorry**

### CODE REVIEW

**Comment by erikcorry** [comment]
This doesn't work - the prints from the logger in jaguar still get mixed up with other output.

To test, run the following on the device with jag monitor and jag run:

```
main:
  foo

foo:
  bar

bar:
  throw "a fish"
```

File: src/os_posix.cc [hunk 1]
```diff
@@ -220,6 +220,7 @@ void OS::set_up() {
   ASSERT(sizeof(void*) == sizeof(pthread_t));
   (void) pthread_key_create(&thread_key, null);
   Thread::ensure_system_thread();
+  _print_mutex = allocate_mutex(-1, "Print mutex");
```
Comment (kasperl) [inline src/os_posix.cc:223]: As discussed, this should probably be 0.

**Review by kasperl — APPROVED** [review]
Looks good to me.

**Comment by kasperl** [comment]
I assume this isn't going in? Feel free to reopen.

**Pull request closed without merging**